motionplan/tpSpaceRRT.go

```diff
@@ -676,16 +646,21 @@ func (mp *tpSpaceRRTMotionPlanner) attemptExtension(
 			}
 		}
 		var err error
-		reseedCandidate, err = mp.extendMap(ctx, candidates, rrt, invert)
+		newReseedCandidate, err := mp.extendMap(ctx, candidates, rrt, isGoalTree)
 		if err != nil && !errors.Is(err, errNoCandidates) {
 			return &nodeAndError{nil, err}
 		}
-		if reseedCandidate == nil {
-			return &nodeAndError{nil, nil}
+		if newReseedCandidate == nil {
+			if reseedCandidate == nil {
```

Comment:
It looks like were doing this twice? But I cant figure out what this one means

Reply:
`reseedCandidate`  is the one from last iteration, that this iteration reseeded with. `newReseedCandidate` is the one we created on this iteration, which if neither are nil will get updated below with `reseedCandidate = newReseedCandidate`.

If both are nil then we return immediately, if only the new one is nil then we want to return the old one.